File: src/main/scala/it/mighe/ssbi/instructions/DecrementPointerInstruction.java
package it.mighe.ssbi.instructions;

import it.mighe.ssbi.Instruction;
import it.mighe.ssbi.Tape;

public class DecrementPointerInstruction extends Instruction {

    public int execute(Tape tape, int programCounter) {
        tape.shiftLeft();
        return programCounter + 1;
    }
}

File: src/main/scala/it/mighe/ssbi/instructions/DecrementPointerInstruction.scala
package it.mighe.ssbi.instructions

import it.mighe.ssbi.Instruction
import it.mighe.ssbi.Tape

class DecrementPointerInstruction extends Instruction {
  def execute(tape: Tape, programCounter: Int): Int = {
    tape.shiftLeft()
    programCounter + 1
  }

  override def execute(tape: Tape): Instruction = {
    tape.shiftLeft()
    next
  }
}